$NetBSD: patch-configure.ac,v 1.3 2020/12/24 01:10:23 manu Exp $

Search for res_ndestroy and use it instead of res_nclose if available

--- configure.ac.orig	2020-09-22 02:55:01.853002338 +0200
+++ configure.ac	2020-09-22 02:55:47.922944675 +0200
@@ -138,8 +138,22 @@
                                            [return res_ninit(NULL);])])
 AC_SEARCH_LIBS(res_ninit, resolv,
 	AC_DEFINE(HAVE_RES_NINIT, 1,
 	[Define to 1 if you have the `res_ninit()' function.]))
+AC_MSG_CHECKING([if res_ndestroy is available])
+AC_LINK_IFELSE([AC_LANG_PROGRAM([
+		#include <sys/types.h>
+		#include <netinet/in.h>
+		#include <arpa/nameser.h>
+		#include <resolv.h>
+	],[
+		struct __res_state res;
+		res_ndestroy(&res);
+	])], [AC_DEFINE([HAVE_RES_NDESTROY], [1],
+	    [we have res_ndestroy]) res_ndestroy=yes],
+	    [res_ndestroy=no])
+AC_MSG_RESULT([$res_ndestroy])
+
 m4_rename_force([saved_AC_LANG_CALL], [AC_LANG_CALL])
 AC_CHECK_LIB(idn, idn_free)
 AC_CHECK_LIB(rt, nanosleep)
 AC_SEARCH_LIBS(inet_addr, nsl)
